bound/kt/src/main/kotlin/tbdex/sdk/http/Exchanges.kt

```diff
@@ -0,0 +1,204 @@
+package tbdex.sdk.http
+
+import tbdex.sdk.messages.*
+import tbdex.sdk.rust.GetExchangeResponseBodyDataSerializedMessage
+import tbdex.sdk.rust.MessageKind
+import tbdex.sdk.rust.CreateExchangeRequestBody as RustCoreCreateExchangeRequestBody
+import tbdex.sdk.rust.GetExchangeResponseBody as RustCoreGetExchangeResponseBody
+import tbdex.sdk.rust.GetExchangeResponseBodyData as RustCoreGetExchangeResponseBodyData
+import tbdex.sdk.rust.GetExchangesResponseBody as RustCoreGetExchangesResponseBody
+import tbdex.sdk.rust.UpdateExchangeRequestBody as RustCoreUpdateExchangeRequestBody
+import tbdex.sdk.rust.ReplyToRequestBody as RustCoreReplyToRequestBody
+import tbdex.sdk.rust.SystemArchitecture
+
+class GetExchangeResponseBody private constructor(
+    val data: List<Message>,
+    private val rustCoreGetExchangeResponseBody: RustCoreGetExchangeResponseBody
+) {
+    init {
+        SystemArchitecture.set() // ensure the sys arch is set for first-time loading
+    }
+
+    constructor(data: List<Message>) : this(
+        data,
+        RustCoreGetExchangeResponseBody(
+            RustCoreGetExchangeResponseBodyData(
+                data.map {
+                    val (kind, jsonSerialized) = when (it) {
+                        is Rfq -> Pair(MessageKind.RFQ, it.toJson())
+                        is Quote -> Pair(MessageKind.QUOTE, it.toJson())
+                        is Order -> Pair(MessageKind.ORDER, it.toJson())
+                        is Cancel -> Pair(MessageKind.CANCEL, it.toJson())
+                        is OrderStatus -> Pair(MessageKind.ORDER_STATUS, it.toJson())
+                        is Close -> Pair(MessageKind.CLOSE, it.toJson())
+                        else -> throw Exception("unknown type $it")
+                    }
+
+                    GetExchangeResponseBodyDataSerializedMessage(kind, jsonSerialized)
+                }
+            )
+        )
+    )
+
+    companion object {
+        fun fromJsonString(json: String): GetExchangeResponseBody {
+            val rustCoreGetExchangeResponseBody = RustCoreGetExchangeResponseBody.fromJsonString(json)
+            val data = rustCoreGetExchangeResponseBody.getData().data.map {
```

Comment:
we have opportunities to improve performance here at a later date, but this is fine for now since this PR is already enormous